templates/code-style-validation.yml

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) 2019 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+jobs:
+# Spin up a Linux VM for validating code.
+- job: code_style_validation
+  displayName: Code Style Validation
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+
+  # Use Python 3 for validating the code.
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: '3.x'
+
+  # Install all the Python code validation packages.
+  - bash: |
+      python -m pip install pre-commit
+      pre-commit run --all
+    displayName: Validate code
+
+  - bash: (test -e docs && tk-docs-preview --build-only) || echo No "docs" folder detected.
```

Comment:
I also added documentation generation. We still never be caught unaware that the documentation does not build just before release now. 😄